grails-app/controllers/org/pih/warehouse/api/StockMovementApiController.groovy

```diff
@@ -193,7 +193,9 @@ class StockMovementApiController {
             if (stockMovement.isDeleteOrRollbackAuthorized(currentLocation)) {
                 if (stockMovement?.isPending() || !stockMovement?.shipment?.currentStatus) {
                     try {
-                        stockMovementService.deleteStockMovement(params.id)
+                        stockMovementService.deleteStockMovement(stockMovement)
+                        stockMovement?.shipment = null
+                        stockMovement?.requisition = null
```

Comment:
let's do it in the service layer, before any `delete()` (GORM's delete) happens (in case anyone forced to add flush there, because it'd break then)

Reply:
what I mean is to add:
```groovy
 stockMovement?.shipment = null
```
 to line 354 (above `shipmentService.deleteShipment`)
and 
```groovy
stockMovement?.requisition = null
```
to line 356 (above `requisitionService.deleteRequisition(stockMovement?.requisition)`)

Reply:
or actually it sucks, because you then pass `stockMovement.requisition` which would be `null` to the `deleteRequisition`, so you should consider creating "a copy" of that object as:
```groovy
Requisition requisition = stockMovement?.requisition
```
use it like that:
```groovy
requisitionService.deleteRequisition(requisition)
```
but above that do:
```groovy
stockMovement?.requisition = null
```